lib/PhantomEnvironment.js

```diff
@@ -390,6 +396,25 @@ PhantomEnvironment.prototype = _.create(Environment.prototype, /**@lends Phantom
             this._on('navigationRequested', function (url) {
                 debug('Navigation to %s', url);
             }),
+            this._on('resourceRequested', /* @covignore */ function (requestData, request, allowedUrls, blockedUrls) {
+                var url = requestData.url;
+                var hasAllowedUrls = Array.isArray(allowedUrls) && allowedUrls.length > 0;
+                var hasBlockedUrls = Array.isArray(blockedUrls) && blockedUrls.length > 0;
+                var allowed = !hasAllowedUrls || allowedUrls.reduce(function (lastValue, urlPattern) {
```

Comment:
you can use [some](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/some) or [every](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/every)